abra_core/src/builtins/native_fns.rs

```diff
@@ -1,14 +1,16 @@
 use crate::typechecker::types::Type;
-// use crate::vm::compiler::fn_name_for_arity;
 use crate::vm::value::{Value, Obj};
 use crate::vm::vm::VMContext;
 use std::collections::HashMap;
 use std::slice::Iter;
+use std::cmp::Ordering;
+use std::fmt::{Debug, Formatter};
 
 // Native functions must return a Value, even if they're of return type Unit.
 // If their return type is Unit, they should return None//Value::Nil.
-type NativeAbraFn = fn(VMContext, Vec<Value>) -> Option<Value>;
+type NativeAbraFn = fn(&VMContext, Vec<Value>) -> Option<Value>;
```

Comment:
This change was a qol change, so we didn't make unnecessary copies of `VMContext` everywhere